model/search/Filter/NotJsonFilter.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\search\Filter;
+
+class NotJsonFilter implements TokenFilterInterface
+{
+    public function filter(string $data): string
+    {
+        $data = trim($data);
+        if (in_array($data[0] ?? '', ['{', '[']) && null !== json_decode($data)) {
```

Comment:
Do you think that encoding a big json to validate would affect too much the performance?

Reply:
IMO - not a bit impact.
This is not a real benchmark, but some numbers to compare

Time to index 40 typical items ( with json filter ), also includes sample with 8Mb json element within item content.
```
 40 resources indexed for class Item (http://www.tao.lu/Ontologies/TAOItem.rdf#Item) by oat\taoAdvancedSearch\scripts\tools\IndexPopulator. offset: 0, limit 100000
  Finished at 2021-06-18T09:04:40+02:00. Indexed 40 resources for class: Item (http://www.tao.lu/Ontologies/TAOItem.rdf#Item).

real	0m1.992s
user	0m0.381s
sys	0m0.262s
```

Same items but without json filter
```
root@b13eac502147:/var/www/html# time php  index.php '\oat\taoAdvancedSearch\scripts\tools\IndexPopulator' -l 100000
Executing:
  40 resources indexed for class Item (http://www.tao.lu/Ontologies/TAOItem.rdf#Item) by oat\taoAdvancedSearch\scripts\tools\IndexPopulator. offset: 0, limit 100000
  Finished at 2021-06-18T09:05:01+02:00. Indexed 40 resources for class: Item (http://www.tao.lu/Ontologies/TAOItem.rdf#Item).

real	0m1.981s
user	0m0.379s
sys	0m0.239s
```

Reply:
Great, indeed it is not big change, specially if you are including 8MB files. Thanks